protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1145,6 +1156,78 @@ func renderServiceTags(services []*descriptor.Service, reg *descriptor.Registry)
 	return tags
 }
 
+// expandPathPatterns search the URI parts for path parameters with pattern and when the pattern contains a sub-path,
+// it expands the pattern into the URI parts and adds the new path parameters to the pathParams slice.
+//
+// Parameters:
+//   - pathParts:	the URI parts parsed from the path template with `templateToParts` function
+//   - pathParams: the path parameters of the service binding, this slice will be mutated when the path pattern contains
+//     a sub-path with wildcard.
+//
+// Returns:
+//
+//	The modified pathParts. Also mutates the pathParams slice.
+func expandPathPatterns(pathParts []string, pathParams *[]descriptor.Parameter) []string {
```

Comment:
I just realized that we have a package for parsing these paths already: https://github.com/grpc-ecosystem/grpc-gateway/blob/main/internal/httprule. What do you think about reusing this?

Reply:
I used that package but unfortunately, it don't fit as a replace for the `templateToParts` function, I encountered two problems

- the `templateToParts` splits the template by slashes, so template with a leading slash creates an empty string at the start of the output slice, trailing slash in the template creates an empty string at the end of the slice - I can go around this, but `httprule.Parse` do not even accept templates without trailing slash, which broke some tests - I guess that all templates should be generally with trailing slash, but I'm not certain.
- the `httprule.Parse` ignores and omits the path patterns, these are completely lost in the `Compile` output object, this was a blocker,
- the `Compile.Pool` contains template segments, but the order is not guaranteed, I encounter a test with a path param where the order of the segments was different than in the original template, this is hard to resolve,

all in all, `httprule.Parse` does a lot of stuff but most of it is not useful for the `templateToParts` and the important stuff is missing and hard to add. At least, I managed to simplify the `templateToParts` and fix the camel-casing, which wrongly affected the parameter pattern.

Reply:
Thanks for the explanation, that makes sense!